compiler/ilgen/TypeDictionary.cpp

```diff
@@ -475,11 +592,25 @@ TypeDictionary::PointerTo(TR::IlType *baseType)
    }
 
 TR::IlReference *
-TypeDictionary::FieldReference(const char *structName, const char *fieldName)
+TypeDictionary::FieldReference(const char *typeName, const char *fieldName)
    {
    TR_HashId structID = 0;
-   _structsByName->locate(structName, structID);
-   StructType *theStruct = (StructType *) _structsByName->getData(structID);
-   return theStruct->getFieldSymRef(fieldName);
+
+   auto found = _structsByName->locate(typeName, structID);
+   if (found)
+      {
+      StructType *theStruct = (StructType *) _structsByName->getData(structID);
+      return theStruct->getFieldSymRef(fieldName);
+      }
+
+   found = _unionsByName->locate(typeName, structID);
+   if (found)
+      {
+      UnionType *theUnion = (UnionType *) _unionsByName->getData(structID);
+      return theUnion->getFieldSymRef(fieldName);
+      }
+
```

Comment:
Hm...

I think the problem here is not what happens when you have the same field name in a struct and a union, but rather what happens when a union and a struct have the same name.

This should of course never happen in a well-formed program, but JitBuilder does not currently check for this. It looks like it doesn't even check for two structs with the same name.

This is a large enough issue that I think it should be solved in a separate pull request.